src/autopas/utils/EnergySensor.cpp

```diff
@@ -0,0 +1,67 @@
+/**
+ * @file EnergySensor.cpp
+ * @author Maximilian Praus
+ * @date 29.06.2024
+ */
+
+#include "autopas/utils/EnergySensor.h"
+
+#include "autopas/options/EnergySensorOption.h"
+#include "pmt.h"
+
+namespace autopas::utils {
+
+EnergySensor::EnergySensor(EnergySensorOption sensor) : _option(sensor) {
+  if (_option != EnergySensorOption::none) {
+    _sensor = pmt::Create(sensor.to_string());
+  } else {
+    AutoPasLog(WARN, "No sensor for energy consumption measurement specified. Energy will not be measured!");
```

Comment:
This shouldn't be a simple `WARN`: If energy is enabled but there is no sensor, then the results used for AutoTuning will be gibberish, right? If so, I would simply throw an exception here.

Reply:
Just to add some input on this:

If I understand the code in the `AutoTuner` correctly, you always instantiate an EnergySensor. The default sensor type is None. Therefore, by default, we would get this warning (or the exception Sam suggested). This is unpleasant.

You could now either 
- only instantiate the `EnergySensor` if the option is not `none` (e.g. wrapping the sensor in `std::optional()`), or 
- always instantiate but only sample energy if the sensor is not `none`.

Either way, I think the most important thing is that we get an exception if someone explicitly wants to measure energy but can't. If nothing is possible but also nothing is requested, I'd prefer not to get a warning.